File: rgctl-tests/ecommerce-go/internal/database/seed.go
package database

import (
	"github.com/google/uuid"
	"github.com/sshaaf/ecommerce-go/internal/models"
	"github.com/sshaaf/ecommerce-go/internal/pkg/timeutil"
	"gorm.io/gorm"
)

func SeedDemo(db *gorm.DB) error {
	var count int64
	if err := db.Model(&models.Category{}).Count(&count).Error; err != nil {
		return err
	}
	if count > 0 {
		return nil
	}

	catID := uuid.NewString()
	cat := models.Category{
		ID:        catID,
		Name:      "Electronics",
		Slug:      "electronics",
		CreatedAt: timeutil.NowISO(),
	}
	if err := db.Create(&cat).Error; err != nil {
		return err
	}

	products := []models.Product{
		{
			ID:          uuid.NewString(),
			CategoryID:  catID,
			Name:        "Wireless Headphones",
			Slug:        "wireless-headphones",
			Description: "Noise cancelling over-ear headphones",
			PriceCents:  12999,
			Stock:       50,
			CreatedAt:   timeutil.NowISO(),
		},
		{
			ID:          uuid.NewString(),
			CategoryID:  catID,
			Name:        "USB-C Hub",
			Slug:        "usb-c-hub",
			Description: "7-in-1 adapter",
			PriceCents:  4999,
			Stock:       120,
			CreatedAt:   timeutil.NowISO(),
		},
	}
	return db.Create(&products).Error
}

File: rgctl-tests/ecommerce-python/app/utils/seed.py
import uuid

from sqlalchemy import func, select
from sqlalchemy.orm import Session

from app.models.category import Category
from app.models.product import Product


def seed_demo_data(db: Session) -> None:
    count = db.scalar(select(func.count()).select_from(Category))
    if count and count > 0:
        return

    category = Category(
        id=str(uuid.uuid4()),
        name="Electronics",
        slug="electronics",
    )
    db.add(category)
    db.flush()

    products = [
        Product(
            id=str(uuid.uuid4()),
            category_id=category.id,
            name="Wireless Headphones",
            slug="wireless-headphones",
            description="Noise cancelling over-ear headphones",
            price_cents=12999,
            stock=50,
        ),
        Product(
            id=str(uuid.uuid4()),
            category_id=category.id,
            name="USB-C Hub",
            slug="usb-c-hub",
            description="7-in-1 adapter",
            price_cents=4999,
            stock=120,
        ),
    ]
    db.add_all(products)
    db.commit()
